src/main/java/com/autotune/analyzer/deployment/AutotuneDeployment.java

```diff
@@ -545,7 +559,12 @@ private static AutotuneConfig getAutotuneConfig(String autotuneConfigResource, K
 
 				Tunable tunable;
 				try {
-					tunable = new Tunable(tunableName, step, upperBound, lowerBound, tunableValueType, queriesMap, sloClassList, layerName);
+					// check the tunable type and then invoke the corresponding constructer 
+					if (tunableValueType.equalsIgnoreCase("categorical")) {
+						tunable = new Tunable(tunableName, step, tunableValueType, queriesMap, sloClassList, layerName, choices);
+					} else {
+						tunable = new Tunable(tunableName, step, upperBound, lowerBound, tunableValueType, queriesMap, sloClassList, layerName);
```

Comment:
Can we have both constructors have common parameters in the same position. This is very confusing to pass the same parameter in different positions depending on if it is a categorical type or regular.

Reply:
Updated as requested